pkg/oc/cli/cmd/image/append/append.go

```diff
@@ -0,0 +1,635 @@
+package append
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"regexp"
+	"runtime"
+	"time"
+
+	units "github.com/docker/go-units"
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	"github.com/docker/distribution"
+	distributioncontext "github.com/docker/distribution/context"
+	"github.com/docker/distribution/manifest/manifestlist"
+	"github.com/docker/distribution/manifest/schema1"
+	"github.com/docker/distribution/manifest/schema2"
+	"github.com/docker/distribution/reference"
+	"github.com/docker/distribution/registry/client"
+	digest "github.com/opencontainers/go-digest"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+
+	"github.com/openshift/origin/pkg/image/apis/image/docker10"
+	imagereference "github.com/openshift/origin/pkg/image/apis/image/reference"
+	"github.com/openshift/origin/pkg/image/dockerlayer"
+	"github.com/openshift/origin/pkg/image/dockerlayer/add"
+	"github.com/openshift/origin/pkg/image/registryclient"
+	"github.com/openshift/origin/pkg/image/registryclient/dockercredentials"
+)
+
+var (
+	desc = templates.LongDesc(`
+		Add layers to Docker images
+
+		Modifies an existing image by adding layers or changing configuration and then pushes that
+		image to a remote registry. Any inherited layers are streamed from registry to registry 
+		without being stored locally. The default docker credentials are used for authenticating 
+		to the registries.
+
+		Layers may be provided as arguments to the command and must each be a gzipped tar archive
+		representing a filesystem overlay to the inherited images. The archive may contain a "whiteout"
+		file (the prefix '.wh.' and the filename) which will hide files in the lower layers. All
+		supported filesystem attributes present in the archive will be used as is.
+
+		Metadata about the image (the configuration passed to the container runtime) may be altered
+		by passing a JSON string to the --image or --meta options. The --image flag changes what
+		the container runtime sees, while the --meta option allows you to change the attributes of
+		the image used by the runtime. Use --dry-run to see the result of your changes. You may
+		add the --drop-history flag to remove information from the image about the system that 
+		built the base image.
+
+		Images in manifest list format will automatically select an image that matches the current
+		operating system and architecture unless you use --filter-by-os to select a different image.
+		This flag has no effect on regular images.
+
+		Experimental: This command is under active development and may change without notice.`)
+
+	example = templates.Examples(`
+# Remove the entrypoint on the mysql:latest image
+%[1]s --from mysql:latest --to myregistry.com/myimage:latest --image {"Entrypoint":null}
```

Comment:
does this get merged w/ the other metadata fields?  or the whole config becomes empty except for Entrypoint?

Reply:
nm looks like it's a patch per the help below.